### PR TITLE
[flow-strict] Flow strict StatusBar

### PR DESCRIPTION
**Pull request by watanabeyu**
Related to #22100

`Libraries/Components/StatusBar/StatusBar.js`
Enhance StatusBar with mergePropsStack and _defaultProps.

## Test Plan:
- [x] npm run prettier
- [x] npm run flow
- [x] npm run flow-check-ios
- [x] npm run flow-check-android
- [x] npm run lint
- [x] npm run test
- [x] ./scripts/run-android-local-unit-tests.sh

## Changelog:
[GENERAL][ENHANCEMENT][StatusBar.js] - apply flow strict-local

### CODE REVIEW

File: Libraries/Components/StatusBar/StatusBar.js [hunk 1]
```diff
@@ -103,13 +103,52 @@ type Props = $ReadOnly<{|
...
+};
+
 /**
  * Merges the prop stack with the default values.
  */
 function mergePropsStack(
-  propsStack: Array<Object>,
-  defaultValues: Object,
-): Object {
+  propsStack: Array<StackEntryProps>,
```
Comment (RSNara) [inline Libraries/Components/StatusBar/StatusBar.js]: We should be able to use `$ReadOnlyArray<StackEntryProps>` here.

Comment (watanabeyu) [reply Libraries/Components/StatusBar/StatusBar.js]: Thanks review.
I changed `Array<StackEntryProps>` to  `$ReadOnlyArray<StackEntryProps>` and `npm run flow` at local, it actually ok.

**Comment by watanabeyu** [comment]
@TheSavior I will try again!

**Comment by watanabeyu** [comment]
@TheSavior I reproduced these error.
`StatusBar.setHidden()` is caused that.
`StatusBar.setHidden()` is required arguments 0.
So I changed arguments 0 required yes -> no.

**Pull request closed without merging**